### ISSUE TITLE
Download list not showing

### ISSUE THREAD

**Issue by kungfoolfighting** [piejanssens/premiumizer#93]
Hello,
I am currently having a problem, where I only see the infinitely spinning wheel when going to the 'Home' page.
The debug log shows a bunch of:
```
12-27 16:33:26       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-27 16:33:26       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-27 16:33:27.626472+01:00 (in 0.996570 seconds)
12-27 16:33:26       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:27 CET)" (scheduled at 2017-12-27 16:33:26.626472+01:00)
12-27 16:33:26       root                                     : DEBUG    : def update started
12-27 16:33:26       root                                     : DEBUG    : def prem_connection started
12-27 16:33:26       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-27 16:33:26       root                                     : DEBUG    : def parse_task started
12-27 16:33:26       apscheduler.executors.default            : ERROR    : Job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:27 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1065, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    task = get_task(transfer['hash'].encode("utf-8"))
KeyError: 'hash'
12-27 16:33:27       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-27 16:33:27       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-27 16:33:28.626472+01:00 (in 0.996913 seconds)
12-27 16:33:27       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:28 CET)" (scheduled at 2017-12-27 16:33:27.626472+01:00)
12-27 16:33:27       root                                     : DEBUG    : def update started
12-27 16:33:27       root                                     : DEBUG    : def prem_connection started
12-27 16:33:27       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-27 16:33:27       root                                     : DEBUG    : def parse_task started
12-27 16:33:27       apscheduler.executors.default            : ERROR    : Job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:28 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1065, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    task = get_task(transfer['hash'].encode("utf-8"))
KeyError: 'hash'
12-27 16:33:28       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-27 16:33:28       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-27 16:33:29.626472+01:00 (in 0.997096 seconds)
12-27 16:33:28       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:29 CET)" (scheduled at 2017-12-27 16:33:28.626472+01:00)
12-27 16:33:28       root                                     : DEBUG    : def update started
12-27 16:33:28       root                                     : DEBUG    : def prem_connection started
12-27 16:33:28       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-27 16:33:28       root                                     : DEBUG    : def parse_task started
12-27 16:33:28       apscheduler.executors.default            : ERROR    : Job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:29 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1065, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    task = get_task(transfer['hash'].encode("utf-8"))
KeyError: 'hash'
12-27 16:33:29       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-27 16:33:29       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-27 16:33:30.626472+01:00 (in 0.997340 seconds)
12-27 16:33:29       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:30 CET)" (scheduled at 2017-12-27 16:33:29.626472+01:00)
12-27 16:33:29       root                                     : DEBUG    : def update started
12-27 16:33:29       root                                     : DEBUG    : def prem_connection started
12-27 16:33:29       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-27 16:33:29       root                                     : DEBUG    : def parse_task started
12-27 16:33:29       apscheduler.executors.default            : ERROR    : Job "update (trigger: interval[0:00:01], next run at: 2017-12-27 16:33:30 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1065, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    task = get_task(transfer['hash'].encode("utf-8"))
KeyError: 'hash'
```

Which also repeats infinitely.

Any idea what is going on here?

**Comment by neox387** [comment]
just noticed this myself, looking into it =D

**Comment by neox387** [comment]
well found out why:

API Change
During an upgrade of our website, the API will also slightly change. Please read the migration guide and find the new API doc at: https://www.premiumize.me/static/api/migration.txt

**Comment by kungfoolfighting** [comment]
ah okay, good to know it is not some configuration or setup issue on my side.
Thanks for looking into it!

**Comment by neox387** [comment]
Well this is a pretty big change, premiumizer replaced the hash of a file with some sort of id, bunch of code needs to be changed.

first PR is up #94

adding a .torrent from watchdir and it downloading seems to work again at least for now, will check again when I have time..

**Comment by kungfoolfighting** [comment]
with the newest version I am getting a slightly different error in the debug log:

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1056, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    add_task(transfer['id'].encode("utf-8"), size, name, '', transfer['folder_id'].encode("utf-8"))
AttributeError: 'NoneType' object has no attribute 'encode'
12-27 23:09:53       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-27 23:09:53       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-27 23:09:54.749701+01:00 (in 0.996221 seconds)
12-27 23:09:53       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:01], next run at: 2017-12-27 23:09:54 CET)" (scheduled at 2017-12-27 23:09:53.749701+01:00)
12-27 23:09:53       root                                     : DEBUG    : def update started
12-27 23:09:53       root                                     : DEBUG    : def prem_connection started
12-27 23:09:53       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-27 23:09:53       root                                     : DEBUG    : def parse_task started
12-27 23:09:53       root                                     : DEBUG    : def get_task started
12-27 23:09:53       apscheduler.executors.default            : ERROR    : Job "update (trigger: interval[0:00:01], next run at: 2017-12-27 23:09:54 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1056, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    add_task(transfer['id'].encode("utf-8"), size, name, '', transfer['folder_id'].encode("utf-8"))
AttributeError: 'NoneType' object has no attribute 'encode'
12-27 23:09:54       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-27 23:09:54       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-27 23:09:55.749701+01:00 (in 0.996911 seconds)
12-27 23:09:54       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:01], next run at: 2017-12-27 23:09:55 CET)" (scheduled at 2017-12-27 23:09:54.749701+01:00)
12-27 23:09:54       root                                     : DEBUG    : def update started
12-27 23:09:54       root                                     : DEBUG    : def prem_connection started
12-27 23:09:54       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-27 23:09:54       root                                     : DEBUG    : def parse_task started
12-27 23:09:54       root                                     : DEBUG    : def get_task started
12-27 23:09:54       apscheduler.executors.default            : ERROR    : Job "update (trigger: interval[0:00:01], next run at: 2017-12-27 23:09:55 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1056, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    add_task(transfer['id'].encode("utf-8"), size, name, '', transfer['folder_id'].encode("utf-8"))
AttributeError: 'NoneType' object has no attribute 'encode'
12-27 23:09:55       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-27 23:09:55       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-27 23:09:56.749701+01:00 (in 0.995531 seconds)
12-27 23:09:55       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:01], next run at: 2017-12-27 23:09:56 CET)" (scheduled at 2017-12-27 23:09:55.749701+01:00)
12-27 23:09:55       root                                     : DEBUG    : def update started
12-27 23:09:55       root                                     : DEBUG    : def prem_connection started
12-27 23:09:55       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-27 23:09:55       root                                     : DEBUG    : def parse_task started
12-27 23:09:55       root                                     : DEBUG    : def get_task started
12-27 23:09:55       apscheduler.executors.default            : ERROR    : Job "update (trigger: interval[0:00:01], next run at: 2017-12-27 23:09:56 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 1056, in update
    idle = parse_tasks(transfers)
  File "premiumizer.py", line 1091, in parse_tasks
    add_task(transfer['id'].encode("utf-8"), size, name, '', transfer['folder_id'].encode("utf-8"))
AttributeError: 'NoneType' object has no attribute 'encode'

```

**Comment by neox387** [comment]
Crap, this is a torrent that is still downloading maybe? And has no folder_id yet meh...

Can fix tomorrow or weekend will see :)

**Comment by kungfoolfighting** [comment]
@neox387 
No there is no download still running, but some downloads have this:
error - NaN KiB/s in the downloads list on premiumize.me

**Comment by neox387** [comment]
Try again after #95 Plz =D

Filehost probably still broken tough

**Comment by kungfoolfighting** [comment]
you mean after the PR is integrated? Or do you want me to manually use the PR branch to test it?

**Comment by neox387** [comment]
Well whatever you want Haha =D

**Comment by piejanssens** [comment]
@kungfoolfighting It's merged, you can test.

**Comment by kungfoolfighting** [comment]
@piejanssens Thanks a bunch. Seems to work again.

**Comment by kungfoolfighting** [comment]
uploading a torrent to premiumize.me now works and it shows the files that are present in the cloud and their download status.
But I cannot download the files to my computer.
The log shows this:

```
12-28 21:45:49: INFO : Task: XXXXXXXXX.torrent -- Category set to: tv
12-28 21:45:53: ERROR : Download failed for: XXXXXXX.torrent, folder_id is None
12-28 21:45:53: WARNING : Retrying failed download in 10 minutes for: XXXXXXXX.torrent
```
Where XXXXXXXX is the torrent name. In the UI I can then see that premiumizer stops trying to download.

**Comment by neox387** [comment]
Browse torrent also does not work then I Guess?
And on the premiumize.me website you can open The folder for the torrent? 
And is this for all torrents?

I added The above error as a precaution :)

**Comment by kungfoolfighting** [comment]
Browse torrent does not work.
I can open the folder under "my files", yes.
It seems to be for all torrents, yes. I have tried 5 different ones now.

**Comment by neox387** [comment]
Mm can't see it, will look on Pc again tomorrow ^^

**Comment by bjsjr** [comment]
I can confirm the issue reported by kungfoolfighting.  I'm receiving the same error and cannot download anything.
All of the Browse buttons for the torrents contain the link 'https://www.premiumize.me/files?folder_id=null'

**Comment by neox387** [comment]
Mm, I had a torrent added with Watchdir and downloaded overnight without issues

**Comment by piejanssens** [comment]
@bjsjr Can you share your premiumizer logs?

**Comment by neox387** [comment]
Well I just noticed a case where there is no folder_id but instead a file_id.. this is a nzb tough

**Comment by kungfoolfighting** [comment]
```
12-29 13:17:09       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/folder/list HTTP/1.1" 200 None
12-29 13:17:09       root                                     : DEBUG    : def processing_dir started
12-29 13:17:09       root                                     : DEBUG    : def clean_name started
12-29 13:17:09       root                                     : DEBUG    : def to_unicode started
12-29 13:17:09       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 13:17:09       apscheduler.executors.downloads          : ERROR    : Job "XXXXXXXXXXXX.torrent (trigger: date[2017-12-29 13:17:09 CET], next run at: 2017-12-29 13:17:09 CET)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
  File "premiumizer.py", line 956, in download_task
    failed = download_process()
  File "premiumizer.py", line 940, in download_process
    process_dir(json.loads(r.content)['content'], greenlet.task.dldir)
  File "premiumizer.py", line 916, in process_dir
    sample = is_sample(x)
  File "premiumizer.py", line 892, in is_sample
    if dir_content['url'].lower().endswith(tuple(media_extensions)):
KeyError: 'url'
12-29 13:17:19       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-29 13:17:19       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-29 13:17:29.653547+01:00 (in 9.986366 seconds)
12-29 13:17:19       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:10], next run at: 2017-12-29 13:17:29 CET)" (scheduled at 2017-12-29 13:17:19.653547+01:00)
12-29 13:17:19       root                                     : DEBUG    : def update started
```

This may be of interest.

**Comment by neox387** [comment]
Ye thanks, another error fixt :D

**Comment by neox387** [comment]
Working on the folder problem, it's getting complicated...

Because if what you upload is only 1 file, premiumize.me doesn't create a folder for it but instead stores it in the root folder :/ so now forced to first create a folder in the premiumize.me cloud and then uploading the torrent/nzb/magnet/filehost to that specific folder sigh .. :d

**Comment by kungfoolfighting** [comment]
umm.. I don't think that is the issue, really.
The torrent I have uploaded contains both a video file and a txt file and already had a folder under "My Files".

**Comment by neox387** [comment]
Ye well try again later it's all related to the folder_id anyway ^^

**Comment by neox387** [comment]
well hit a wall...

premiumize.me api docs:
Target folder is not supported, but the parameter will be named "folder_id"

So the function to start a new transfer directly in a folder isn't there yet..

**Comment by kungfoolfighting** [comment]
Is it not possible to individually download the files?

**Comment by neox387** [comment]
@piejanssens Do you know how this works:

old api docs:

> Move items into the root folder from somewhere else
Use https://www.premiumize.me/api/folder/paste?items=itemarray
Pass an array named "items" containing the elements to move.
itemarray: This should be a multidimensional array with items, containing two name fields. The first is "type", the other one is "id". Those values can be obtained from the folder list.

> Move items into a folder from somewhere else
Use https://www.premiumize.me/api/folder/paste?id=targetfolderid&items=itemarray 
You need to pass the id of the folder where you want to move to as paramter "id", the itemarray is constructed like above.

new api:

> Also the /api/folder/paste method has changed to accept two arrays of IDs
(files and folders) instead of one multidimensional array with id and type.

> Form parameters
files (optional)
Form Parameter — items to add to the zip
folders (optional)
Form Parameter — folders to add to the zip
target_id (required)
Form Parameter — id of the folder to paste into

I can't get this to work :/

The premiumize.me website still uses the old method it seems copy/paste results in parameters:
id:mkCqALSvdUD_P69RSBCJvQ                             this is the folder where to paste in
items[0][id]:-jo6xQG68mpY3iA93TmV3g               this is the file_id of what to paste
items[0][type]:folder

How do you make items[0][id] in python ? oO there is no array on lists mm ? :d

**Comment by kungfoolfighting** [comment]
The api doc says it will still support the old way for a while. So perhaps just leave it as it is, so you can make it is usable for now. Then when it works, you can work on updating that API interaction.

**Comment by neox387** [comment]
Noop wont work transfer/list doesn't respond with a hash etc anymore,
if old api still worked there wouldn't have been any issue with loading tasks anyway right :)

**Comment by kungfoolfighting** [comment]
Well as far as I can tell, the issue is, that torrents are now no longer listed as torrents when you list the root folder. They are listed as files. So that needed to be fixed. But the paste API has merely been deprecated, but not outright changed, according to the migration doc.
Other than that, what exactly did you have to change?
I am looking at the premiumizer.py right now, dusting off my Python. Once I get an overview of the functionality, perhaps I can help.

**Comment by neox387** [comment]
never used the paste API before, only need it now to put a single root file into a folder, because premiumizer is build for gettings urls (that are now called links by the api...) from the folder of that transfer.

but now a new transfer doesn't automatically get it's own folder anymore if it's just 1 file.
Best solution would be start a new transfer into folder X but this isn't implemented yet.. as the old docs say:
Target folder is not supported, but the parameter will be named "folder_id".

so have to let it finish the download into root(if it doesn't get a folder aka multiple files), and then create a proper folder and paste the file into the folder.


Anyway got that far, only prob is donno howto make a post request with an array ..

**Comment by piejanssens** [comment]
@neox387 So this issue is currently only for single files? If so, what's the issue with Premiumize downloading it in the cloud root?

**Comment by kungfoolfighting** [comment]
https://stackoverflow.com/questions/33541631/how-to-send-multidimensional-post-in-python
This url covers using the request lib.
I saw the premiumizer.py also has an import for request.

**Comment by neox387** [comment]
@piejanssens 
Trying something ^^

**Comment by neox387** [comment]
#96 not using the paste stuff etc just added the file_id and let it download to root..

**Comment by neox387** [comment]
@kungfoolfighting thanks for link that was probably The way :D

Anyway try again with latest version.

**Comment by bjsjr** [comment]
This seems to have fixed the problem for me. Everything that was queued up is now downloading.

Thanks guys!

**Comment by neox387** [comment]
Great :)

Just noticed I removed few to many things, will look into adding torrent magnet and file host from home page later..

**Comment by bjsjr** [comment]
Sorry but I spoke too soon.  I've come across a couple of problems now that I have some more items to download.

It looks like premiumizer isn't assigning the categories of new items.  Here's a snippet form the log showing the item being added:

`2-29 23:25:01       watchdog.observers.inotify_buffer        : DEBUG    : in-event <InotifyEvent: src_path=blackhole/tv/XXXX.S01E09.WEB-DL.x264-RARBG.torrent, wd=3, mask=IN_CREATE, cookie=0, name=XXXX.S01E09.WEB-DL.x264-RARBG.torrent>
12-29 23:25:01       watchdog.observers.inotify_buffer        : DEBUG    : in-event <InotifyEvent: src_path=blackhole/tv/XXXX.S01E09.WEB-DL.x264-RARBG.torrent, wd=3, mask=IN_MODIFY, cookie=0, name=XXXX.S01E09.WEB-DL.x264-RARBG.torrent>
12-29 23:25:01       watchdog.observers.inotify_buffer        : DEBUG    : in-event <InotifyEvent: src_path=blackhole/tv/XXXX.S01E09.WEB-DL.x264-RARBG.torrent, wd=3, mask=IN_ATTRIB, cookie=0, name=XXXX.S01E09.WEB-DL.x264-RARBG.torrent>
12-29 23:25:01       watchdog.observers.inotify_buffer        : DEBUG    : in-event <InotifyEvent: src_path=blackhole/tv/XXXX.S01E09.WEB-DL.x264-RARBG.torrent, wd=3, mask=IN_ATTRIB, cookie=0, name=XXXX.S01E09.WEB-DL.x264-RARBG.torrent>
12-29 23:25:06       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-29 23:25:06       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-29 23:25:16.129227-05:00 (in 9.993827 seconds)
12-29 23:25:06       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:10], next run at: 2017-12-29 23:25:16 EST)" (scheduled at 2017-12-29 23:25:06.129227-05:00)
12-29 23:25:06       root                                     : DEBUG    : def update started
12-29 23:25:06       root                                     : DEBUG    : def prem_connection started
12-29 23:25:06       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-29 23:25:06       root                                     : DEBUG    : def parse_task started
12-29 23:25:06       root                                     : DEBUG    : def get_task started
12-29 23:25:06       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:06       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"running","file_id":null,"speed":"  --- ","dlsize":"  --- ","local_status":null,"category":"movie","name":"XXXXX (2017) [720p] [YTS.AG]","id":"NXyAKFgxa5ZbUbfasnJi9w","eta":"Loading...","progress":0,"folder_id":null}}]
12-29 23:25:06       root                                     : DEBUG    : def get_task started
12-29 23:25:06       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:06       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"running","file_id":null,"speed":" 0 mbit/s from 0 peers --- ","dlsize":"260.5 MB --- ","local_status":null,"category":"","name":"XXXX.S01E01.WEB-DL.x264-RARBG","id":"XSYmqDnU87GgQtXVxqY_1w","eta":" unknown","progress":9,"folder_id":null}}]
12-29 23:25:06       root                                     : DEBUG    : def get_task started
12-29 23:25:06       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:06       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"queued","file_id":null,"speed":" 0.61 mbit/s from 1 peers --- ","dlsize":"247.6 MB --- ","local_status":null,"category":"","name":"XXXX.S01E10.WEB-DL.x264-RARBG","id":"GvGafM-7lv1dwjmnqmeGpQ","eta":" 0:38:30","progress":15,"folder_id":null}}]
12-29 23:25:06       socketio                                 : INFO     : emitting event "tasks_updated" to all [/]
12-29 23:25:06       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["tasks_updated",{}]
12-29 23:25:06       apscheduler.executors.default            : INFO     : Job "update (trigger: interval[0:00:10], next run at: 2017-12-29 23:25:16 EST)" executed successfully
12-29 23:25:06       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-29 23:25:06       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-29 23:25:16.264390-05:00 (in 9.998060 seconds)
12-29 23:25:11       root                                     : DEBUG    : New file detected at: blackhole/tv/XXXX.S01E09.WEB-DL.x264-RARBG.torrent
12-29 23:25:11       root                                     : DEBUG    : def upload_torrent started
12-29 23:25:11       root                                     : DEBUG    : Uploading torrent to the cloud: blackhole/tv/XXXX.S01E09.WEB-DL.x264-RARBG.torrent
12-29 23:25:11       root                                     : DEBUG    : def prem_connection started
12-29 23:25:11       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/create HTTP/1.1" 200 None
12-29 23:25:11       root                                     : DEBUG    : Upload successful: blackhole/tv/XXXX.S01E09.WEB-DL.x264-RARBG.torrent
12-29 23:25:16       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run
12-29 23:25:16       apscheduler.scheduler                    : DEBUG    : Next wakeup is due at 2017-12-29 23:25:26.264390-05:00 (in 9.989520 seconds)
12-29 23:25:16       apscheduler.executors.default            : INFO     : Running job "update (trigger: interval[0:00:10], next run at: 2017-12-29 23:25:26 EST)" (scheduled at 2017-12-29 23:25:16.264390-05:00)
12-29 23:25:16       root                                     : DEBUG    : def update started
12-29 23:25:16       root                                     : DEBUG    : def prem_connection started
12-29 23:25:16       urllib3.connectionpool                   : DEBUG    : https://www.premiumize.me:443 "POST /api/transfer/list HTTP/1.1" 200 None
12-29 23:25:16       root                                     : DEBUG    : def parse_task started
12-29 23:25:16       root                                     : DEBUG    : def get_task started
12-29 23:25:16       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:16       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"running","file_id":null,"speed":"  --- ","dlsize":"  --- ","local_status":null,"category":"movie","name":"XXXX (2017) [720p] [YTS.AG]","id":"NXyAKFgxa5ZbUbfasnJi9w","eta":"Loading...","progress":0,"folder_id":null}}]
12-29 23:25:16       root                                     : DEBUG    : def get_task started
12-29 23:25:16       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:16       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"running","file_id":null,"speed":" 0 mbit/s from 0 peers --- ","dlsize":"260.5 MB --- ","local_status":null,"category":"","name":"XXXX.S01E01.WEB-DL.x264-RARBG","id":"XSYmqDnU87GgQtXVxqY_1w","eta":" unknown","progress":9,"folder_id":null}}]
12-29 23:25:16       root                                     : DEBUG    : def get_task started
12-29 23:25:16       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:16       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"queued","file_id":null,"speed":" 0.27 mbit/s from 1 peers --- ","dlsize":"247.6 MB --- ","local_status":null,"category":"","name":"XXXX.S01E10.WEB-DL.x264-RARBG","id":"GvGafM-7lv1dwjmnqmeGpQ","eta":" 0:48:29","progress":16,"folder_id":null}}]
12-29 23:25:16       root                                     : DEBUG    : def get_task started
12-29 23:25:16       root                                     : DEBUG    : def add_task started
12-29 23:25:16       root                                     : DEBUG    : def get_task started
12-29 23:25:16       root                                     : DEBUG    : def get_cat_var started
12-29 23:25:16       root                                     : INFO     : Added: XXXX.S01E09.WEB-DL.x264-RARBG -- Category: 
12-29 23:25:16       root                                     : DEBUG    : def get_task started
12-29 23:25:16       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:16       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"queued","file_id":null,"speed":" 0 mbit/s from 0 peers --- ","dlsize":"259.1 MB --- ","local_status":null,"category":"","name":"XXXX.S01E09.WEB-DL.x264-RARBG","id":"VowENd6OgNOllL7rxQycfg","eta":" unknown","progress":0,"folder_id":null}}]
12-29 23:25:16       socketio                                 : INFO     : emitting event "update_task" to all [/]
12-29 23:25:16       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["update_task",{"task":{"cloud_status":"queued","file_id":null,"speed":" 0 mbit/s from 0 peers --- ","dlsize":"259.1 MB --- ","local_status":null,"category":"","name":"XXXX.S01E09.WEB-DL.x264-RARBG","id":"VowENd6OgNOllL7rxQycfg","eta":" unknown","progress":0,"folder_id":null}}]
12-29 23:25:16       socketio                                 : INFO     : emitting event "tasks_updated" to all [/]
12-29 23:25:16       engineio                                 : INFO     : 4c158b261dde47dba89a50d7433a4ede: Sending packet MESSAGE data 2["tasks_updated",{}]
12-29 23:25:16       apscheduler.executors.default            : INFO     : Job "update (trigger: interval[0:00:10], next run at: 2017-12-29 23:25:26 EST)" executed successfully
12-29 23:25:16       apscheduler.scheduler                    : DEBUG    : Looking for jobs to run`

Another issue is that when the torrent is added to the cloud queue it's not removing the .torrent file from the blackhole directory and keeps trying to add it again as seen in the log snippet:

`12-29 23:32:13: ERROR : Upload of torrent: blackhole/tv/XXXX.S01E10.WEB-DL.x264-RARBG.torrent failed, message: You already have this job added.`

**Comment by neox387** [comment]
Another PR #97 Should hopefully be ok now ..

**Comment by neox387** [comment]
@kungfoolfighting @bjsjr I assume everything is working again ? :)

**Comment by bjsjr** [comment]
Yes, for the most part.  Thank you very much!

I recently switched over to jdownloader and premiumizer is not always removing the file from the cloud after dl.  It seems to be sometimes, but not others.  Here's a snippet from the log after the file has been dl'd.

Since I just started using jd2 I'm not sure if this is related to the changes you recently made or if it's unrelated.

```
01-01 14:16:08: INFO : Downloading: XXX.XXX.2005.720p.BluRay.H264.AAC-RARBG
01-01 14:16:09: INFO : File not downloaded it already exists at: /mnt/movienfs/XXX.XXX.2005.720p.BluRay.H264.AAC-RARBG/XXX.XXX.2005.720p.BluRay.H264.AAC-RARBG.mp4
01-01 14:22:48: WARNING : Download: XXX.XXX.2005.720p.BluRay.H264.AAC-RARBG stuck restarting task
01-01 14:22:52: INFO : Downloading: XXX.XXX.2005.720p.BluRay.H264.AAC-RARBG
01-01 14:22:53: INFO : File not downloaded it already exists at: /mnt/movienfs/XXX.XXX.2005.720p.BluRay.H264.AAC-RARBG/XXX.XXX.2005.720p.BluRay.H264.AAC-RARBG.mp4

```

**Comment by neox387** [comment]
Mm,
stuck restarting task
Is a check I added to see if a task that is supposed to be downloading is stuck based on the size that is downloaded compared to the size 1 minute later

File not downloaded it already exist, explains itself.. 

So normally that task should just complete cause there is nothing to Download?

But if it doesn't remove remove from the cloud after the task is done it probably errored some where and might not be visible without debug mode. 

Ill check The code later this week hadnt had an issue with this :)

In next PR I added aria2 as a new Download client :)

**Comment by kungfoolfighting** [comment]
@neox387 
Sorry, for inactivity on my part, the new years festivities have kept me busy.
The only thing I have tried so far was to set the category to TV for some downloads that were already in my cloud and the download worked fine.
So, I would say for me the issue is closed for now. If I find further problems, I will open another issue.
Thanks once again for your work!
